### DISCUSSION TITLE
[Apps] pass Teleport username as a Header to an App

### DISCUSSION THREAD

**Discussion by Esa-mimbias** [gravitational/teleport#25549] [Q&A] [answered]
Hi

on several documentations and examples, we discover how to pass `logins` to a client Web Application with, for example:
```
  rewrite:
    headers:
       - "X-Internal-Trait: {{internal.logins}}"
```
The App receive an header with the login the user used to connect.

I'd like to pass Teleport username, for example to fill the login field of a legacy app (i.e. non JWT, non OIDC, non SAML); but I can't manage to find the correct attribute (the one I'd use between {{ }} )

Furthermore, I couldn't find a way to list those attributes : `internal.logins`, `internal.jwt` are mentioned in the docs, what are the others ? [Doc](https://goteleport.com/docs/application-access/guides/connecting-apps/) says "Rewritten header values support the same templating variables as [role templates](https://goteleport.com/docs/access-controls/guides/role-templates/)" but it's not clear for me how to get those variables. 
I also tried to have a look in the source code, but It seems I'm not skilled enough in reading Go.

Any clue ?
Thank you

**Comment by webvictim** [comment, marked as answer]
There isn't an easy way to do this with local users currently. There's a feature request here: https://github.com/gravitational/teleport/issues/17616
